src/shaders/_prelude_fog.fragment.glsl

```diff
@@ -2,9 +2,24 @@
 
 uniform vec2 u_fog_range;
 uniform vec3 u_fog_color;
+uniform vec3 u_haze_color_linear;
 uniform float u_fog_opacity;
+uniform float u_fog_exponent;
 uniform float u_fog_sky_blend;
 uniform float u_fog_temporal_offset;
+uniform float u_haziness;
+
+// A tone map is a smooth min between x and 1.0
+vec3 tonemap (vec3 color) {
+    // Exponential smooth min
+    const float k = 8.0;
+    return -log2(exp2(-k * color) + exp2(-k)) * (1.0 / k);
+
+    // Cubic smooth min (works well)
+    //const float k = 0.8;
+    //vec3 h = max(k - abs(color - 1.0), vec3(0)) / k;
+    //return max(vec3(0), min(color, vec3(1)) - h * h * h * k * (1.0 / 6.0));
```

Comment:
nit: Remove extraneous comment (can be left in PR description for archiving).
```suggestion

    // Cubic smooth min (works well)
    //const float k = 0.8;
    //vec3 h = max(k - abs(color - 1.0), vec3(0)) / k;
    //return max(vec3(0), min(color, vec3(1)) - h * h * h * k * (1.0 / 6.0));
```

Reply:
The functions are almost visually indistinguishable. Thought I might double-check if we have a good way to micro-benchmark before definitely removing.